ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -606,6 +602,36 @@ public void getTableBranches(Context ctx) {
     ctx.json(OkResponse.of(amsPageResult));
   }
 
+  /**
+   * cancel the running optimizing process of one certain table.
+   *
+   * @param ctx - context for handling the request and response
+   */
+  public void cancelOptimizingProcess(Context ctx) {
+    String catalog = ctx.pathParam("catalog");
+    String db = ctx.pathParam("db");
+    String table = ctx.pathParam("table");
+    String processId = ctx.pathParam("processId");
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(catalog)
+            && StringUtils.isNotBlank(db)
+            && StringUtils.isNotBlank(table),
+        "catalog.database.tableName can not be empty in any element");
+    Preconditions.checkState(tableService.catalogExist(catalog), "invalid catalog!");
+
+    ServerTableIdentifier serverTableIdentifier =
+        tableService.getServerTableIdentifier(
+            TableIdentifier.of(catalog, db, table).buildTableIdentifier());
+    TableRuntime tableRuntime = tableService.getRuntime(serverTableIdentifier);
```

Comment:
serverTableIdentifier maybe null if the table doesn't have a TableRuntime.

Reply:
> serverTableIdentifier maybe null if the table doesn't have a TableRuntime.

Done, plz review it again, thx.